contrib/babelfishpg_tsql/src/catalog.c

```diff
@@ -2751,6 +2803,60 @@ get_partition_depend_table_oid(HeapTuple tuple, TupleDesc dsc)
 	return ObjectIdGetDatum(table_oid);
 }
 
+static Datum
+get_pivot_view_dbname(HeapTuple tuple, TupleDesc dsc)
+{
+	bool		is_null;
+	char		*dbname;
+	Datum		dbid = heap_getattr(tuple, Anum_bbf_pivot_view_dbid, dsc, &is_null);
+
+	if (is_null) /* Sanity check. */
+		ereport(ERROR,
+				(errcode(ERRCODE_NULL_VALUE_NOT_ALLOWED),
+					errmsg("dbid should not be null in babelfish_pivot_view catalog")));
+	
+	/* Another way to check for existence of dbid in babelfish_sysdatabases catalog. */
+	dbname = get_db_name(DatumGetInt16(dbid));
+
+	if (!dbname) /* Sanity check. */
+		ereport(ERROR,
+				(errcode(ERRCODE_INTERNAL_ERROR),
+					errmsg("dbid in babelfish_pivot_view catalog doesn't exists in babelfish_sysdatabases catalog")));
+
+	return CStringGetTextDatum(dbname);
+}
+
+static Datum
+get_pivot_view_schema_name(HeapTuple tuple, TupleDesc dsc)
+{
+	bool		schema_is_null, dbid_is_null;
+	char		*physical_schema_name, *schema_name, *org_schema_name;
```

Comment:
nit: it's good practice to initialize variables and not trust the functions such as `heap_getattr()` to initialize them.

Reply:
all vars are initialized to `NULL` in latest revision